### PR TITLE
Configurable products stock calculation improvement 

### PR DESCRIPTION
**Pull request by jnmrclmbsjse**
This PR includes:
1. Fix for configurable products showing out of stock in UI even though variations are orderable.
2. Ran formatter

Scenarios:
https://app.clickup.com/t/8693q167z?comment=90120023271172&threadedComment=90120024905985

### CODE REVIEW

**Comment by jnmrclmbsjse** [comment]
Task linked: [CU-8693q167z Products say they're out-of-stock with Orderability on Always](https://app.clickup.com/t/8693q167z)

File: tests/unit/Imports/ProductImportTest.php [hunk 1]
```diff
@@ -155,6 +157,156 @@ function (MockInterface $module) {
...
+                'simple',
+                true,
+                false,
+                5,
+                10,
+            ],
+            true,
+            true,
+            10,
+        ];
```
Comment (ikulis) [inline tests/unit/Imports/ProductImportTest.php:175]:
```suggestion
        $tests['simple or configurable assign product in stock, limited, has positive orderable value'] = [
            [
                'type' => 'simple',
                'product_stock' => [
                      'in_stock' => true,
                      'unlimited' => false,
                      'value' => 5,
                ] ,
                'orderable_stock_value' => 10,
            ],
            [
                    'in_stock' => true,
                    'manage' => true,
                    'quantity' => 5,
              ]
        ];
```

Comment (jnmrclmbsjse) [reply tests/unit/Imports/ProductImportTest.php]: Changes implemented.

File: tests/unit/Imports/ProductImportTest.php [hunk 2]
```diff
@@ -155,6 +157,194 @@ function (MockInterface $module) {
...
+                'orderable_stock_value' => null,
+            ],
+            [
+                'in_stock' => true,
+                'manage_stock' => true,
+                'quantity' => 5,
+            ],
+        ];
+
+        $tests['simple or configurable assign product out stock, limited, has orderable value'] = [
```
Comment (ikulis) [inline tests/unit/Imports/ProductImportTest.php:215]: that on is a strange one, you got it back from backned?

File: tests/unit/Imports/ProductImportTest.php [hunk 3]
```diff
@@ -155,6 +157,194 @@ function (MockInterface $module) {
...
+                'product_stock' => [
+                    'in_stock' => false,
+                    'unlimited' => true, // Ignored cause in_stock is false
+                    'value' => 5,
+                ],
+                'orderable_stock_value' => 10, // Ignored cause in_stock is false
+            ],
+            [
+                'in_stock' => false,
+                'manage_stock' => true,
```
Comment (ikulis) [inline tests/unit/Imports/ProductImportTest.php:244]:
```suggestion
                'manage_stock' => false,
```
it should be false here, because unlimited

File: tests/unit/Imports/ProductImportTest.php [hunk 4]
```diff
@@ -155,6 +157,194 @@ function (MockInterface $module) {
         $this->assertCount(1, $variableProduct->get_upsell_ids(), '1 upsell IDs should be set');
     }
 
+    public function dataProviderTestGetStockProperties()
```
Comment (ikulis) [inline tests/unit/Imports/ProductImportTest.php:160]: I made lately https://github.com/storekeeper-company/api-helper-scripts/blob/master/src/ExportChannelProductsToWooCommerceFormatCommand.php#L47 which contains all the logic. 
I needed it for compasison

File: tests/unit/Imports/ProductImportTest.php [hunk 5]
```diff
@@ -155,6 +157,194 @@ function (MockInterface $module) {
...
+            [
+                'in_stock' => true,
+                'manage_stock' => true,
+                'quantity' => 10,
+            ],
+        ];
+
+        // Configurable products are forced to be in stock and unlimited if orderable value is < 0, otherwise it will show out of stock
+        // in shop https://app.clickup.com/t/8693q167z, it should then depend on the variations if they can be ordered or not
+        $tests['configurable product in stock, limited, has negative orderable value'] = [
```
Comment (ikulis) [inline tests/unit/Imports/ProductImportTest.php:268]: I though configurable always will aggregate stock from it's variations in woocomrce. Any idea why we set it manually?

Comment (jnmrclmbsjse) [reply tests/unit/Imports/ProductImportTest.php]: Changed tests.

File: tests/unit/Imports/ProductImportTest.php (hunk 4 above)
Comment (jnmrclmbsjse) [reply tests/unit/Imports/ProductImportTest.php:160]: Changes made based on scenarios tested and discussed. See PR description for details.

File: tests/unit/Imports/ProductImportTest.php (hunk 2 above)
Comment (jnmrclmbsjse) [reply tests/unit/Imports/ProductImportTest.php]: Changed tests.

File: tests/unit/Imports/ProductImportTest.php [hunk 6]
```diff
@@ -155,6 +157,209 @@ function (MockInterface $module) {
         $this->assertCount(1, $variableProduct->get_upsell_ids(), '1 upsell IDs should be set');
     }
 
+    public function dataProviderTestGetStockProperties()
+    {
+        $tests = [];
+
+        $tests['simple or configurable assign product in stock, limited, has 0 orderable stock'] = [
```
Comment (ikulis) [inline tests/unit/Imports/ProductImportTest.php:164]:
```suggestion
        $tests['simple  product in stock, limited, has 0 orderable stock'] = [
```

can you also change other test names ?

File: tests/unit/Imports/ProductImportTest.php [hunk 7]
```diff
@@ -155,6 +157,209 @@ function (MockInterface $module) {
         $this->assertCount(1, $variableProduct->get_upsell_ids(), '1 upsell IDs should be set');
     }
 
+    public function dataProviderTestGetStockProperties()
```
Comment (ikulis) [inline tests/unit/Imports/ProductImportTest.php:160]: should also add backorderable checking to this unit test? that we have all covered. Or is there another test where it's covered

Comment (jnmrclmbsjse) [reply tests/unit/Imports/ProductImportTest.php:160]: Added backorders test

File: tests/unit/Imports/ProductImportTest.php (hunk 6 above)
Comment (jnmrclmbsjse) [reply tests/unit/Imports/ProductImportTest.php]: Corrected test names

File: tests/unit/Imports/ProductImportTest.php [hunk 8]
```diff
@@ -329,6 +346,178 @@ public function dataProviderTestGetStockProperties()
...
+                    'value' => -5,
+                ],
+                'orderable_stock_value' => -5,
+                'backorder_enabled' => true,
+            ],
+            [
+                'in_stock' => true, // always in stock because backorders are allowed
+                'manage_stock' => true,
+                'quantity' => 0,
+                'backorders' => 'yes',
```
Comment (ikulis) [inline tests/unit/Imports/ProductImportTest.php:384]:
```suggestion
                'in_stock' => false,
                'manage_stock' => true,
                'quantity' => 0,
                'backorders' => 'yes', 
```

File: tests/unit/Imports/ProductImportTest.php [hunk 9]
```diff
@@ -329,6 +346,178 @@ public function dataProviderTestGetStockProperties()
...
+
+        $tests['configurable, value negative or 0, limited, backorder enabled'] = [
+            [
+                'type' => 'configurable',
+                'product_stock' => [
+                    'in_stock' => false,
+                    'unlimited' => false,
+                    'value' => -5,
+                ],
+                'orderable_stock_value' => -5,
```
Comment (ikulis) [inline tests/unit/Imports/ProductImportTest.php:472]:
```suggestion
                'orderable_stock_value' => 0,
```
lets do 0 here because edge case
